flutter_app/lib/main.dart

```diff
@@ -1,125 +1,81 @@
 
-  // This widget is the root of your application.
-  @override
-  Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        // This is the theme of your application.
-        //
-        // TRY THIS: Try running your application with "flutter run". You'll see
-        // the application has a blue toolbar. Then, without quitting the app,
-        // try changing the seedColor in the colorScheme below to Colors.green
-        // and then invoke "hot reload" (save your changes or press the "hot
-        // reload" button in a Flutter-supported IDE, or press "r" if you used
-        // the command line to start the app).
-        //
-        // Notice that the counter didn't reset back to zero; the application
-        // state is not lost during the reload. To reset the state, use hot
-        // restart instead.
-        //
-        // This works for code too, not just values: Most code changes can be
-        // tested with just a hot reload.
-        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
-        useMaterial3: true,
-      ),
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
-    );
-  }
-}
-
-class MyHomePage extends StatefulWidget {
-  const MyHomePage({super.key, required this.title});
+  parser.stream.listen((MavlinkFrame frm) {
+    if (frm.message is CommandLong) {
+      var cl = frm.message as CommandLong;
 
-  // This widget is the home page of your application. It is stateful, meaning
-  // that it has a State object (defined below) that contains fields that affect
-  // how it looks.
+      if (cl.command == 512) {
+        print("Received Request Message");
+        print("Message id: ${cl.param1}");
+      } else if (cl.command == 511) {
+        print("Received Set Message Interval");
+        print("Interval: ${cl.param2} μs");
+        print("Message id: ${cl.param1}");
+      }
+    }
+  });
 
-  // This class is the configuration for the state. It holds the values (in this
-  // case the title) provided by the parent (in this case the App widget) and
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
+  var sequence = 0;
+  var systemId = 255;
+  var componentId = 1;
 
-  final String title;
+  // example of requesting a single instance of a message using MAV_CMD_REQUEST_MESSAGE (512)
+  var desiredMessage = 33; // GLOBAL_POSITION_INT
+  var param2 =
+      0; // parameter required for some messages, described in common message set
+  var commandLong = CommandLong(
+      targetSystem: 1,
+      targetComponent: 0,
+      command: 512,
+      confirmation: 0,
+      param1: desiredMessage.toDouble(),
+      param2: param2.toDouble(),
+      param3: 0,
+      param4: 0,
+      param5: 0,
+      param6: 0,
+      param7: 0);
+  var frm = MavlinkFrame.v1(sequence, systemId, componentId, commandLong);
+  parser.parse(frm.serialize());
```

Comment:
Is this working with Mission Planner?

Reply:
I haven't tested it with mission planner yet. Can mission planner receive messages over TCP?